ext/ddtrace_profiling_native_extension/native_extension_helpers.rb

```diff
@@ -12,8 +12,8 @@ module NativeExtensionHelpers
       # Can be set to force rubygems to fail gem installation when profiling extension could not be built
       ENV_FAIL_INSTALL_IF_MISSING_EXTENSION = 'DD_PROFILING_FAIL_INSTALL_IF_MISSING_EXTENSION'
 
-      # Older Rubies don't have the MJIT header, used by the JIT compiler, so we need to use a different approach
-      CAN_USE_MJIT_HEADER = RUBY_VERSION >= '2.6'
+      # The MJIT header was introduced on 2.6 and removed on 3.3; for other Rubies we rely on debase-ruby_core_source
+      CAN_USE_MJIT_HEADER = RUBY_VERSION.start_with?('2.6', '2.7', '3.0.', '3.1.', '3.2.')
```

Comment:
Why do you add a last `.` for version 3 and above? 

What do you think of `RUBY_VERSION > '2.6' && RUBY_VERSION <= '3.2'`?

Reply:
I'm adding a `.`because technically in around 7 years maybe Ruby `3.10` will come around and `3.10` would match `3.1` as well. But I'm not particularly consistent in doing this everywhere.

> What do you think of RUBY_VERSION > '2.6' && RUBY_VERSION <= '3.2'?

No strong opinions, if you think it's worth it, I can change it.

Reply:
Actually we do:

```[1] pry(main)> ruby_version = "3.10.0"
=> "3.10.0"
[2] pry(main)> ruby_version > '2.6' && ruby_version <= '3.2'
=> true
```

Reply:
As a last suggestion, which is not at all necessary, we could use `Gem::Version` instance.

```ruby
ruby_version = Gem::Version.new("3.10.0")
=> Gem::Version.new("3.10.0")
irb(main):007:0> ruby_version > '2.6' && ruby_version <= '3.2'
=> false
```

Reply:
We talked about this on slack, but let me copy this here for posteriority:

> Even though the simplified version has potential for bugs, my thinking is:
> 
> * We never got to `.10` in the past on a Ruby release, that I know of (at least minor version -- patch version yes, like 2.4.10)
> * We set the maximum version that can be used in the gem, so it's not like dd-trace-rb 1.11 is ever going to be used with Ruby 3.10
> * If ever Ruby 3.8/3.9 approaches, we can do a search in the entire codebase by `RUBY_VERSION` and fix this anyway, it's not like it's _that_ hard to find (and would be caught by tests in a lot of cases)